lib/v8.js

```diff
@@ -78,16 +78,24 @@ const { getOptionValue } = require('internal/options');
  * @param {string} [filename]
  * @param {{
  *   exposeInternals?: boolean,
- *   exposeNumericValues?: boolean
+ *   exposeNumericValues?: boolean,
+ *   path?: string,
+ *   encoding?: string
  * }} [options]
  * @returns {string}
  */
 function writeHeapSnapshot(filename, options) {
   if (filename !== undefined) {
     filename = getValidatedPath(filename);
   }
+
+  if (options?.path !== undefined) {
+    options.path = getValidatedPath(options.path);
+  }
+
   const optionArray = getHeapSnapshotOptions(options);
-  return triggerHeapSnapshot(filename, optionArray);
+  const encoding = options?.encoding || 'utf8';
+  return triggerHeapSnapshot(filename, optionArray, options?.path, encoding);
```

Comment:
Nit: Defaulting this here isn't necessary; `ParseEncoding` will fall back to the provided default if the value is undefined or not a string.

```suggestion
  return triggerHeapSnapshot(filename, optionArray, options?.path, options?.encoding);
```